test/test-pipe-set-fchmod.c

```diff
@@ -0,0 +1,30 @@
+#include "uv.h"
+#include "task.h"
+
+TEST_IMPL(pipe_set_fchmod) {
+  uv_pipe_t pipe_handle;
+  uv_loop_t* loop;
+  int r;
+
+  loop = uv_default_loop();
+
+  r = uv_pipe_init(loop, &pipe_handle, 0);
+  ASSERT(r == 0);
+
+  r = uv_pipe_bind(&pipe_handle, TEST_PIPENAME);
+  ASSERT(r == 0);
+
+  /* No easy way to test if this works, we will only make sure that */
+  /* the call is successful. */
+  r = uv_pipe_chmod(&pipe_handle, UV_READABLE);
```

Comment:
on Mac I've used this chmod(pipe_server_name, 0777);
As a temporary work around.
But you'll need the file name